libminifi/include/core/state/Value.h

```diff
@@ -46,26 +56,99 @@ class Value {
     return string_value;
   }
 
-  bool empty(){
+  template<typename T>
+  bool convertValue(T &ref) {
+    return convertValueImpl<typename std::common_type<T>::type>(ref);
+  }
+
+  bool empty() {
     return string_value.empty();
   }
 
+  std::type_index getTypeIndex() {
+    return type_id;
+  }
+
+  static const std::type_index UINT64_TYPE;
+  static const std::type_index INT64_TYPE;
+  static const std::type_index INT_TYPE;
+  static const std::type_index BOOL_TYPE;
+  static const std::type_index STRING_TYPE;
+
  protected:
-  std::string string_value;
 
+  template<typename T>
+  bool convertValueImpl(T &ref) {
+    return getValue(ref);
+  }
+
+  template<typename T>
+  void setTypeId() {
+    type_id = std::type_index(typeid(T));
+  }
+
+  virtual bool getValue(int &ref) {
+    ref = std::stol(string_value);
+    return true;
+  }
+
+  virtual bool getValue(int64_t &ref) {
+    ref = std::stoll(string_value);
+    return true;
+  }
+
+  virtual bool getValue(uint64_t &ref) {
+    ref = std::stoull(string_value);
```

Comment:
These throw. What's the intent here?

Are we happy with an unhandled exception in case the getvalue call fails or shall we catch and set return value based on that?

Reply:
Yeah, and that's handled by the call path. Logs should make that evident when run in these cases (hopefully).